### PR TITLE
Raise Http404 if BlogCategory is not found

### LINKED ISSUE #402: Server error when accessing blogs of non-existing category
Hi,
I've the following setup
Django==1.9.4
django-cms==3.2.5
djangocms-blog==0.7.0

When I try to access /category/test (here test is a non-existing category) there's a server error. Further debugging the issue, I think that it is generated from the djangocms_blog module. Here's the debug log to confirm that.
```
Traceback:

File "/home/pcoder/pyvenv/myproject/lib/python3.6/site-packages/django/core/handlers/base.py" in get_response
  149.                     response = self.process_exception_by_middleware(e, request)

File "/home/pcoder/pyvenv/myproject/lib/python3.6/site-packages/django/core/handlers/base.py" in get_response
  147.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/home/pcoder/pyvenv/myproject/lib/python3.6/site-packages/cms/utils/decorators.py" in inner
  17.         return func(request, *args, **kwargs)

File "/home/pcoder/pyvenv/myproject/lib/python3.6/site-packages/django/views/generic/base.py" in view
  68.             return self.dispatch(request, *args, **kwargs)

File "/home/pcoder/pyvenv/myproject/lib/python3.6/site-packages/aldryn_apphooks_config/mixins.py" in dispatch
  18.         return super(AppConfigMixin, self).dispatch(request, *args, **kwargs)

File "/home/pcoder/pyvenv/myproject/lib/python3.6/site-packages/django/views/generic/base.py" in dispatch
  88.         return handler(request, *args, **kwargs)

File "/home/pcoder/pyvenv/myproject/lib/python3.6/site-packages/djangocms_blog/views.py" in get
  182.             self.request.toolbar.set_object(self.category)

File "/home/pcoder/pyvenv/myproject/lib/python3.6/site-packages/djangocms_blog/views.py" in category
  175.                 get_language(), slug=self.kwargs['category']

File "/home/pcoder/pyvenv/myproject/lib/python3.6/site-packages/django/db/models/query.py" in get
  387.                 self.model._meta.object_name

Exception Type: DoesNotExist at /en-us/cms/blog/category/6/
Exception Value: BlogCategory matching query does not exist.
```
**Expected result**: If the BlogCategory does not exist, I would expect the module to raise an `Http404`

### PR DESCRIPTION
**Pull request by pcoder**
Tentative fix for error #402  

Fix #402

### CODE REVIEW

**Comment by yakky** [comment]
@pcoder LGTM.
Could you rebase on current develop (or merge develop on top of your branch) to port the isort fix?

File: tests/test_views.py [hunk 1]
```diff
@@ -484,3 +484,14 @@ def test_templates(self):
...
+    def test_non_existing_blog_category_should_raise_404(self):
+        pages = self.get_pages()
+        with smart_override('en'):
+            request = self.get_request(pages[1], 'en', AnonymousUser())
+            view_obj = CategoryEntriesView()
+            view_obj.request = request
+            view_obj.namespace, view_obj.config = get_app_instance(request)
+            with self.assertRaises(Http404):
+                view_obj.kwargs = {'category': 'unknown-category'}
+                category_obj = view_obj.get_object()
```
Comment (yakky) [inline tests/test_views.py]: As a `ListView` `CategoryEntriesView` does not have `get_object` method. You can call `get_queryset` which triggers the `category` property where you fixed the error

**Comment by pcoder** [comment]
@yakky Done, could you review again?

**Comment by yakky** [comment]
@pcoder thanks! merging this!

**Pull request merged**